src/main/java/podz/task/Deadline.java

```diff
@@ -0,0 +1,42 @@
+package podz.task;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+
+    protected String by, formattedBy, time;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+
+        String dateAndTime[] = by.split(" ");
```

Comment:
Perhaps the array specifiers should be attached to the type instead of the variable.